src/common/modals/AddInstance/InstanceName.js

```diff
@@ -124,6 +124,28 @@ const InstanceName = ({
         );
       }
 
+      const isForgeModpack = (manifest?.minecraft?.modLoaders || []).some(
+        v => v.id.includes(FORGE) && v.primary
+      );
+
+      const isFabricModpack = (manifest?.minecraft?.modLoaders || []).some(
+        v => v.id.includes(FABRIC) && v.primary
+      );
+
+      if (isForgeModpack) {
+        version.loaderType = FORGE;
```

Comment:
Don't directly mutate the state, use setVersion instead

Reply:
I will admit, my React experience in very minimal, so bear with me.

The control flow is going to get really weird here. If I call `setVersion` that will only queue up the state change, which is weird as I'm supposed to creating the instance _now_. If I don't handle it now, then we'd either need fetch the manifest again, or I'd need to add the manifest itself to the state as well.

The workaround I've thought of is:

  1. At the start of `createInstance` create a copy of the `version` state.
  2. When I discover the loader type from the manifest, update the copied state
  3. Update all references to `version` within `createInstance` to refer to the copied state
  4. At the end of `createInstance` call `setVersion` with the copied state. (Maybe I don't need to do this?)

Does this seem reasonable?

Reply:
You're right, the state change would be queued. I still don't like mutating the state but I guess it's acceptable in this specific situation.